web-client/src/components/AddSockForm.tsx

```diff
@@ -0,0 +1,266 @@
+                                        key={size}
+                                        value={size}
+                                        disabled={
+                                            selectedSizes.includes(size) && field.size !== size
+                                        }
+                                    >
+                                        {size}
+                                    </option>
+                                ))}
+                            </select>
+
+                            <Input
+                                {...register(`sizes.${index}.quantity`, {
+                                    required: "Quantity is required",
+                                    valueAsNumber: true,
+                                    min: 1,
+                                })}
+                                placeholder="Quantity"
+                                type="number"
+                                className="border p-2 text-black"
+                            />
+
+                            <Input
+                                {...register(`sizes.${index}.price`, {
+                                    required: "Price is required",
+                                    valueAsNumber: true,
+                                    min: 0.01,
+                                })}
+                                placeholder="Price"
+                                type="number"
+                                className="border p-2 text-black"
+                            />
+
+                            <Button
+                                size="icon"
+                                variant="destructive"
+                                onClick={() => remove(index)}
+                                className="h-8 w-8 flex items-center justify-center bg-red-500 border-red-500 text-white"
+                            >
+                                <Trash2 className="h-4 w-4" />
+                            </Button>
+                        </div>
+                    ))}
+
+                    <Button
+                        type="button"
+                        variant="outline"
+                        className="w-full border-black text-black"
+                        onClick={() => append({ size: "S", quantity: 1, price: 0 })}
+                    >
+                        + Add Size
+                    </Button>
+                </div>
+            )}
+
+
+            {step === 3 && (
+                <div className="space-y-4">
+                    <label className="block text-sm font-medium text-black">
+                        Preview Image URL
+                    </label>
+                    <Input
+                        {...register("previewImageUrl", {
+                            required: "Image URL is required",
+                            pattern: {
+                                value: /^(https?:\/\/)?[\w.-]+(?:\.[\w\.-]+)+[\w\-._~:/?#[\]@!$&'()*+,;=.]+$/,
```

Comment:
Regexes are quite dangerous espeically when there are long and complex like this. I would say just simplify to ensure a valid url starts with either `https://` OR `http://`

This way, the regex is WAY simpler. Since we are using `Zod` for type validation, you can define that this field is a valid URL I believe, but your current approeach is fine for MVP.